check/configuration/custom-linters/reference.md

```diff
@@ -0,0 +1,527 @@
+version from its runtime or use the default version from the download.
+
+### `version_command`
+
+`version_command`:  Version check commands
+
+### `enabled`
+
+`enabled`: *optional boolean*. Whether this linter is enabled.
+
+### `batch`
+
+`batch`: *optional boolean*. Combine multiple files into the same execution.
+
+### `run_when`
+
+`run_when`, Should this linter be run only in CI? (or locally, vscode, monitor)
+
+### `in_place`
+
+`in_place`: *optional boolean*. Applies to formatters - does the formatter rewrite the file as 
+opposed to reading its contents from STDIN and outputting formatted contents to STDOUT?
+
+### `run_timeout`
+
+`run_timeout`: *duration string*. Describes how long a linter can run before timing out. 
+[See timeouts](../readme.md#timeout)
+
+### `disabled`
+
+`disabled`: *optional boolean*: Whether linter is actively disabled (and will not be recommended)
+and will not run (overrides enabled).
+
+### `commands`
+`commands`: Commands exposed by this linter. [See Linter Command Definition](#linter-command-definition)
+
+### `deprecated`
+
+`deprecated`: *string*. Deprecated information (set when linter is deprecated)
+
+### `known_bad_versions`
+
+`known_bad_versions`: *string list*. Versions of a linter that are known to be broken 
+and should not be run with Trunk. We will fall back to a `known_good_version` 
+if init or upgrade chooses something in this set.
+
+### `plugin_url`
+
+`plugin_url`: *string*, a plugin url for reporting issues.
+
+### `query_compile_commands`
+
+`query_compile_commands`, *optional boolean*
+
+### `idempotent`
+
+`idempotent`: *optional boolean*. Indicates whether a linter is idempotent w.r.t. config + 
+source code inputs.  e.g. semgrep fetches rules from the internet, so it's not idempotent . If 
+set, causes cache_ttl to default to 24h
+
+### `cache_ttl`
+
+`cache_ttl`, duration string, How long cached results are kept before they expire
+
+### `target`
+
+`target`: optional string, what target does this run on
+
+### `suggest_if`
+
+How to determine if this linter should be auto-enabled/recommended. Possible 
+values are `never`, `config_present`, and `files_present`.
+
+* `config_present` will auto-enable a linter if Trunk sees any [`direct_config`](#direct_configs) for it .
+* `files_present` will auto-enable a linter if Trunk sees any file type that it operates on.
+* `never` will never auto-enable this linter.
+
+Trunk curates the values of `suggest_if` for all linters in the [plugins](https://github.com/trunk-io/plugins) repo.
+
+### `supported_platforms`
+
+Platform constraint. If incompatible, renders a notice.
+
+### `tools`
+
+`tools`, string list
+
+### `main_tool`
+
+`main_tool`, *string*. The linter's main tool. This should be the tool that provides the binary 
+for the linter. It will generally be the same as the linter's name, but may differ if the linter 
+is a wrapper. The main tool is used to determine the linter's version.
```

Comment:
```suggestion
`main_tool`, *string*. If your linter depends on more than a single tool, and none of the tools has the same name as the linter, then you will need to specify which is the main tool here. It will be used to version the tool from the linter's enabled version.
```